addons/account/static/src/components/document_file_uploader/document_file_uploader.js

```diff
@@ -0,0 +1,77 @@
+import { useService } from "@web/core/utils/hooks";
+import { FileUploader } from "@web/views/fields/file_handler";
+import { standardWidgetProps } from "@web/views/widgets/standard_widget_props";
+
+import { Component, markup } from "@odoo/owl";
+
+export class DocumentFileUploader extends Component {
+    static template = "account.DocumentFileUploader";
+    static components = {
+        FileUploader,
+    };
+    static props = {
+        ...standardWidgetProps,
+        record: { type: Object, optional: true },
+        divClass: { type: String, optional: true },
+        slots: { type: Object, optional: true },
+        resModel: { type: String, optional: true },
+    };
+
+    setup() {
+        this.orm = useService("orm");
+        this.action = useService("action");
+        this.notification = useService("notification");
+        this.attachmentIdsToProcess = [];
+        this.extraContext = this.getExtraContext();
+    }
+
+    // To pass extra context while creating record
+    getExtraContext() {
+        return {};
+    }
+
+    async onFileUploaded(file) {
+        const att_data = {
+            name: file.name,
+            mimetype: file.type,
+            datas: file.data,
+        };
+        const [att_id] = await this.orm.create("ir.attachment", [att_data], {
+            context: { ...this.extraContext, ...this.env.searchModel.context },
+        });
+        this.attachmentIdsToProcess.push(att_id);
+    }
+
+    // To define specific resModal from another model
+    getResModel() {
+        return this.props.resModel;
+    }
+
+    async onUploadComplete() {
+        const resModal = this.getResModel();
+        const action = await this.orm.call(resModal, "create_document_from_attachment", ["", this.attachmentIdsToProcess], {
+            context: { ...this.extraContext, ...this.env.searchModel.context },
+        });
+        this.attachmentIdsToProcess = [];
+        if (action.context && action.context.notifications) {
+            for (let [file, msg] of Object.entries(action.context.notifications)) {
+                this.notification.add(
+                    msg,
+                    {
+                        title: file,
+                        type: "info",
+                        sticky: true,
+                    });
+            }
+            delete action.context.notifications;
+        }
+        if (action.help && typeof(action.help) === "string") {
```

Comment:
```suggestion
        if (action.help?.length) {
```
I think this will do the same.

Also, if I'm not mistaken, we just want non empty strings, this condition is not very clear about that imo.